src/server/auth/server/api_server.go

```diff
@@ -1774,6 +1774,9 @@ func (a *apiServer) generateAndInsertAuthTokenNoTTL(ctx context.Context, subject
 
 // generates a token, and stores it's hash and supporting data in postgres
 func (a *apiServer) insertAuthToken(ctx context.Context, tokenHash string, subject string, ttlSeconds int64) error {
+	if _, err := a.env.DB.ExecContext(ctx, `INSERT INTO auth.principals (subject, first_seen) VALUES ($1, $2) ON CONFLICT DO NOTHING;`, subject, time.Now()); err != nil {
```

Comment:
I considered it, but didn’t think it was necessary since nothing deletes from `auth.principals`. But of course, something _could_, so I think you’re right after all.